### PR TITLE
Fix infinite scroll

### PR DESCRIPTION
**Pull request by Osmose**
Fixes the infinite scroll to count lines properly (it was previously counting number of files returned) and to append new lines for a file that is already on the page to that file instead of creating a new file group in the search results.

This supersedes #384, which was very helpful in helping me understand what was going wrong. :D

### CODE REVIEW

**Comment by Osmose** [comment]
@erikrose r?

File: dxr/static/js/dxr.js [hunk 1]
```diff
@@ -140,7 +140,7 @@ $(function() {
         requestsInFlight = 0,  // Number of search requests in flight, so we know whether to hide the activity indicator
         displayedRequestNumber = 0,
         didScroll = false,
-        resultCount = 0,
+        resultsLineCount = 0,
```
Comment (erikrose) [inline dxr/static/js/dxr.js]: Can you remind me what this is even for? :-)

Comment (Osmose) [inline dxr/static/js/dxr.js]: From what I can tell, `previousDataLimit` is the amount of lines returned by the last request to the API (which is determined by the amount of lines we can show on a single screen, I think). If `resultsLineCount` is equal to `previousDataLimit` then presumably we got as many lines as we requested from the API, and there may be more (or we might be "lucky" and asked for just the right amount and hit the end of the list.

So it's used to determine if we need to fetch any more results when an infinite scroll request is triggered. I don't think it's the best way to determine that but I also just want to make it work for now, and refactor it down later.

In a related note, I'm not entirely sure the math is working out exactly right anyway, as locally infinite scroll seems to trigger too early, but I couldn't see where the issue was and figured it wasn't a blocker as it doesn't affect functionality much.

File: dxr/static/js/dxr.js [hunk 2]
```diff
@@ -339,8 +350,31 @@ $(function() {
...
+                    .append(nunjucks.render('partial/results_container.html', data));
+            } else {
+                var resultsList = contentContainer.find('.results');
+
+                // If the first result is already on the page (meaning we showed
+                // some of its lines but just fetched more), add new lines to
+                // it instead of adding a new result section.
+                var firstResult = data.results[0];
+                var domFirstResult = resultsList.find(
+                    '.result[data-path="' + firstResult.path + '"]');
```
Comment (erikrose) [inline dxr/static/js/dxr.js]: I'm going to make a few comments just for "here's something to think about" purposes, but they shouldn't block merging this.

Comment one: this is probably a linear search, depending on what kind of data structures the browser uses. We could speed it up, if we need to, by squirreling away the path of the last result in a known place somewhere (or even just looking it up `data-path` on the last result and comparing against that).

**Comment by erikrose** [comment]
This isn't quite right. I can make it do all kinds of weird things, like cutting off (and printing the footer) before all the matches are found, or showing matches 1-73 and then skipping straight to 148-152. Repros to come, hopefully.

**Comment by erikrose** [comment]
1. Put 152 lines of identical text containing "her" into an indexed file, and build the index.
2. Search for "her", pressing Return.
3. Scroll to the bottom.
4. Type in "her" again, not pressing Return.
5. Scroll to the bottom. Only matches 1-73 are found, and then the footer is shown. There should be 152 in all.

**Comment by Osmose** [comment]
I believe I've found the issue; `previousDataLimit` and `dataOffset` were being updated within `doQuery` when a new query was being typed and the query wasn't long enough to trigger a search. Also `dataOffset` wasn't properly being reset to 0 when a new search was happening.

**Comment by erikrose** [comment]
Goodie: I can't get the lines to skip anymore.

Baddie: Now the lines link to just `//source/#25`, leaving out the tree name and thus not working.

**Comment by Osmose** [comment]
Ah, so nunjucks doesn't support `{% from 'foo.html' import blah with context %}`. Updated to explicitly pass the context variables we need (namely www_root and the tree name).

**Comment by erikrose** [comment]
Yeah, it's safer that way anyway. Otherwise, it's effectively dynamic scoping, which is really hard to reason about in any program of size.

**Comment by erikrose** [comment]
It's still not working for me, even after re-running `make`:
screen shot 2015-04-21 at 16 49 38

**Comment by erikrose** [comment]
r+!

**Pull request merged**